aten/src/ATen/native/native_functions.yaml

```
@@ -14175,6 +14175,9 @@
 
 - func: linalg_solve(Tensor A, Tensor B, *, bool left=True) -> Tensor
   python_module: linalg
+  dispatch:
+    CPU, CUDA: linalg_solve
```

This change would break the dispatcher composability (for example I think with these dispatch keys autograd is broken because there's no rule for `linalg_solve`, only for `_linalg_solve_ex`:
https://github.com/pytorch/pytorch/blob/313eec02cc6bbdf3310801cfc04d750ba5539eef/tools/autograd/derivatives.yaml#L1562-L1565
`linalg_solve` is defined here:
https://github.com/pytorch/pytorch/blob/005ae01d6ae4a57af21bc3b8edf2ca21e5fed73d/aten/src/ATen/native/BatchLinearAlgebra.cpp#L1934-L1939

I think it should be marked as `CompositeImplicitAutograd`, a little bit of description of what it does can be found here https://github.com/pytorch/pytorch/blob/005ae01d6ae4a57af21bc3b8edf2ca21e5fed73d/aten/src/ATen/native/README.md#dispatch
```suggestion
    CompositeImplicitAutograd: linalg_solve
```

---

I tried setting `CompositeImplicitAutograd` dispatch key for `linalg_solve` function. However, it caused the `SparseCsrCUDA: _sparse_csr_linear_solve` dispatch rule no longer effective. In other words, all `torch.linalg.solve` function calls with CUDA Sparse CSR tensor inputs are also being routed to `at:: linalg_solve`. Any workaround on this?

---

@zitongzhan Hi Zitong, I see that you have book an OH with us (PT Dev Infra) this Friday (July 5th).  However, please keep in mind that we could only help unblock CI issues on your PR.  A technical question like the one you have here should better be answered by experts like @IvanYashchuk and @eqy

---

Resolved via [6853fa6](https://github.com/pytorch/pytorch/pull/129856/commits/6853fa60a830bdf4019a1b3d7a1f4547163c5879)